sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+
+The identity of the Cloud Admin with access to Management Cluster can be spoofed
+by other users via stolen cloud and kubernetes credentials since cloud provider
+credentials are stored or are able to retrived from the workload cluster.
+
+For on-prem infrastructures, the Cloud Admin will have provisioned the cluster
+with wide ranging permissions, including the ability to create and delete VMs,
+add and remove disks etc... These by default will be written to the CPI
+credentials file on the disk of the workload control plane instances. A user of
+the cluster can start a pod with a host mount of /etc, and exfiltrate the cloud
+provider credentials.
+
+For hyperscaler environments like AWS, there is the ability to scope short lived
+credentials to the level of EC2 instances. However, the permissions are still
+broad for CPI and CSI to function, and a management cluster hosted in AWS will
+have even more permissions, such as the ability to create and destroy EC2
+instances. A user of the cluster can start a pod, and scrape the IMDS endpoint
+living on link-local (`http://169.254.169.254`) and retrieve these credentials.
+They are short-lived but a confused deputy attack is still possible.
+
+Cloud Admin has access to multiple Cloud services like EC2, ELB, Secrets
+Manager. Access to the management cluster basically provides this admin access
+to all the managed workload clusters and its workloads as well.
+
+#### STRIDE-SPOOF-1 Recommended mitigations
+
+Securely manage Cloud and Kubernetes Credentials with following methods applied:
+
+* Credentials are short-lived
+  * Status: Implemented (AWS)
+* Credentials are managed centrally from an authoritative identity store
+  * Status: Implemented (AWS & Azure)
+* Credentials can be revoked, if needed
+  * Status: Implemented (AWS & Azure)
+* Second pair of eyes are applied when executing privileged actions such as
+  creating/deleting/updating a cluster
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+* Ensure credentials used by Cluster API are least privileges and setting access
+  control on Cluster API controller namespaces to prevent unauthorised access by
+  anyone other than cloud admin.
+  - Status: Implemented for K8s RBAC, developer guidance to be written for
+    providers
+* Enable auditing on the management cluster as well as on the cloud provider
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-SPOOF-2 Spoofing of cloud provider services
+
+Cloud provider services like Secrets Manager store sensitive data like cloudinit
+configuration files
+(shortlived). Other services like EC2 and ELB API have the ability to sniff
+traffic destined to the API server of the workload cluster. Successful spoofing
+of a cloud service could result in information disclosure and tampering of in
+flight requests relevant to cluster and machine management.
+
+#### STRIDE-SPOOF-2 Recommended Mitigations
```

Comment:
Same here (I'm not really sure I understand the action required here 😅 )

Reply:
No Action needed. This mostly informational